print-lisp.c

```diff
@@ -0,0 +1,247 @@
+/*
+ * Copyright (c) 2015 Ritesh Ranjan (r.ranjan789@gmail.com)
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ * 3. The name of the author may not be used to endorse or promote products
+ *    derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE AUTHOR ``AS IS'' AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED.  IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/*
+ * tcpdump filter for LISP - Locator/Identifier Separation Protocol
+ * RFC 6830
+ */
+
+#ifdef HAVE_CONFIG_H
+#include "config.h"
+#endif
+
+#include <tcpdump-stdinc.h>
+#include <string.h>
+#include <stdlib.h>
+
+#include "ip.h"
+#ifdef INET6
+#include "ip6.h"
+#endif
+
+#include "extract.h"
+#include "interface.h"
+#include "addrtoname.h"
+
+#define IPv4_AFI 1
+#define IPv6_AFI 2
+
+static const char *type_name[] = {
+	"Reserved",
+	"Map-Request",
+	"Map-Reply",
+	"Map-Register",
+	"Map-Notify",
+	"Unallocated-type",
+	"Unallocated-type",
+	"Unallocated-type",
+	"Control-Message",
+};
+
```

Comment:
A better way:
```
#define LISP_MAP_REQUEST            1
#define LISP_MAP_REPLY              2
#define LISP_MAP_REGISTER           3
#define LISP_MAP_NOTIFY             4
#define LISP_ENCAPS_CONTROL_MESSAGE 8
static const struct tok type_name[] = {
        { 0,                           "Reserved"        },
        { LISP_MAP_REQUEST,            "Map-Request"     },
        { LISP_MAP_REPLY,              "Map-Reply"       },
        { LISP_MAP_REGISTER,           "Map-Register"    },
        { LISP_MAP_NOTIFY,             "Map-Notify"      },
        { LISP_ENCAPS_CONTROL_MESSAGE, "Control-Message" },
        { 0, NULL }
};
```
With this tok structure, you can print the `type_name` with possible error with `tok2str` function:
`tok2str(type_name, "unknown-type-%u", type)`